server/src/main/java/org/opensearch/index/reindex/ReindexRequest.java

```diff
@@ -307,7 +308,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.startObject("source");
             if (remoteInfo != null) {
                 builder.field("remote", remoteInfo);
-                builder.rawField("query", remoteInfo.getQuery().streamInput(), RemoteInfo.QUERY_CONTENT_TYPE.type());
+                builder.rawField("query", remoteInfo.getQuery().streamInput(), (XContentType) RemoteInfo.QUERY_CONTENT_TYPE.mediaType());
```

Comment:
Shouldn't we have `writeRawField` overloaded method which accepts medua type? (the same you did for `public void writeRawValue(InputStream stream, MediaType mediaType) throws IOException` in `JsonXContentGenerator`

Reply:
yes! good catch.